### DISCUSSION TITLE
Prototype 4 (Raspberry Pi hat PCB)

### DISCUSSION THREAD

**Discussion by thijstriemstra** [corrados/edrumulus#55] [General]
I see a start for a 4th prototype has been started in the repository's main branch @corrados

Awesome!

Will it be an easy upgrade for users of v3? Any schematic changes? What is the plan for v4?

**Comment by corrados** [comment]
The new prototype is a Raspberry Pi Hat design similar to [Exadrumino Nano Every](https://github.com/SpintroniK/exadrumino-Nano-Every). Here is an early model:
![grafik](https://user-images.githubusercontent.com/46655886/180230297-fcedcb37-61a8-4217-9ca6-f460665ccfd7.png)
It uses a Roland Multi-Core instead of separate trigger input jacks.
The schematic have not changed in this prototype.

**Comment by corrados** [comment]
Prototype 4 is progressing:
![grafik](https://user-images.githubusercontent.com/46655886/182040937-1c039899-08cc-42e8-9a82-8e8dc414b66b.png)

**Comment by corrados** [comment]
The new prototype 4 is ready:
![grafik](https://user-images.githubusercontent.com/46655886/182689374-81676209-5449-412a-ad12-83ac79469f9f.png)
See the [ChangeLog](https://github.com/corrados/edrumulus/blob/main/doc/ChangeLog.md#2022-08-03-edrumulus-prototype-4) for details.

**Reply by thijstriemstra** [reply]
looks great!! Is the prototype 3 codebase compatible with prototype 4? Can/did you tag the old code?

**Reply by corrados** [reply]
The codebase is compatible to both prototypes. Yon can select it with a compile flag: https://github.com/corrados/edrumulus/blob/cf4f507c6d4aef890a6f4a83acb3bd38df01dee0/edrumulus.ino#L19

**Reply by thijstriemstra** [reply]
Thanks!

**Comment by thijstriemstra** [comment]
I like the new design but it looks the Roland cable plugging into the new prototype is at least 60 euros, plus I can't re-use my existing cables. A set of 8 jack female sockets is about $15 and with some resistors it looks like a cheaper option to me. Also, the Raspberry Pi is currently in high demand and quite pricey. So to me it's a different pricepoint compared to the prototype v3 and because this project is 'competing' with other cheap edrum solutions out there I hope it'll stay compatible with the budget v3 implementations one could make :)

At the end of the day, if Roland releases a quality module that matches the price of a 'hobbyist' (my own soldering, wiring, your awesome code) implementation, what would you choose?

**Reply by corrados** [reply]
I am always try to be as cheap as possible with my prototypes. Since I now try to develop a Raspberry Pi hat, the space on the PCB is very limited so I cannot use normal jack sockets. The prototype 4 can be usd without a Raspberry Pi, too. You can still connect the ESP32 board via USB to the PC.

BTW, I am working in a prototype 5 which will not have a USB socket. With that prototype a Raspberry Pi will be mandatory.

After that, I'll eventually create a new PCB which is much larger and uses normal jack sockets and also normal sized components (i.e., no SMD parts). Would you be interested in such a PCB?

**Reply by corrados** [reply]
To answer your last question: As written in my other forum thread, the cheapest Roland module with positional sensing support is over 1000 €.

But anyway, I am just learning hardware design with my prototypes right now :-). I had never designed a PCB before and I have never soldered SMD parts before. The first PCB had one design problem (see the ChangeLog). I hope that the designs are getting better and better. I am not intending to break compatibility to old hardware designs with future software versions.

**Reply by thijstriemstra** [reply]
Thanks for the feedback and your vision for edrumulus. 

> Would you be interested in such a PCB?

For sure! I'd love to make my prototype a lot smaller by having a pcb. If you end up creating something like that, my email is in the git log and I'll send you money (paypal?) for these costs and shipping; let me know how it works, I've not ordered a custom pcb before). I also saw you had oldskool MIDI in/out female sockets planned, that would be much appreciated as well.

> the cheapest Roland module with positional sensing support is over 1000 €.

Ah, I didn't research it properly and thought it was more in area of 600 euro. But that 1000 euro pricetag has to come down at some point, Moore's law etc right?! :-)

Keep up the great work!

**Reply by corrados** [reply]
> For sure! I'd love to make my prototype a lot smaller by having a pcb.

Ok, good to know. It will take some time to work on this since I first want to finish the other prototype. I'll keep you informed.

**Reply by thijstriemstra** [reply]
> Ok, good to know. It will take some time to work on this since I first want to finish the other prototype. I'll keep you informed.

Perhaps a new discussion topic can be started when you're ready for that; feel free to tag me there for feedback.

**Reply by corrados** [reply]
> I like the new design but it looks the Roland cable plugging into the new prototype is at least 60 euros, plus I can't re-use my existing cables. A set of 8 jack female sockets is about $15 and with some resistors it looks like a cheaper option to me.

Just one more comment regarding the prices. The entry level Roland drum kits all use the DB25 connector.
In your case you already have the cables available. If you start from nothing, you'll have:
1. With DB25:
- DB25 connector: [0.70 €](https://www.reichelt.de/ch/de/d-sub-stecker-25-polig-gewinkelt-rm-9-4-d-sub-st-25eu-p7002.html?r=1)
- Roland multi core cable: [49 €](https://www.thomann.de/de/roland_td_9_td_11_td_15_multicore.htm)

2. With jack sockets:
- jack sockets: 8 x [0.47 €](https://www.reichelt.de/ch/de/klinkeneinbaubuchse-6-3-mm-stereo-ebs-63p-p7303.html?&trstct=pos_2&nbc=1) = 3.76 €
- kabels: 8 x [8.90 €](https://www.thomann.de/de/roland_v_drum_cable_2m.htm) = 71.20 €

So, it is actually more expensive not to use the DB25 cables. I assume you can save some money if you buy the parts for the cables and solder them yourself. But at the same time, one could find out the cable assignments of the cheap [Millenium MPS-850 Cable Loom](https://www.thomann.de/de/millenium_mps_850_cable_loom.htm) and redesign the Edrumulus DB25 connector to support this product instead of the Roland cables. This would be half the price. But since Roland sets standards for the e-drums, it makes sense to develop the support for their cables in Edrumulus.

**Reply by corrados** [reply]
> After that, I'll eventually create a new PCB which is much larger and uses normal jack sockets and also normal sized components (i.e., no SMD parts).

Here is the PCB I have created:
![grafik](https://user-images.githubusercontent.com/46655886/187967002-a69bd048-a243-47a1-a0d3-db77a5027321.png)
It is called prototype 6.

**Comment by thijstriemstra** [comment]
> kabels: 8 x [8.90 €](https://www.thomann.de/de/roland_v_drum_cable_2m.htm) = 71.20 €

wowow, hold on :-) We should really use ebay/aliexpress for these comparisons. reichelt and thomas are 99% of the time the most expensive options out there.

**Reply by corrados** [reply]
> the Roland cable plugging into the new prototype is at least 60 euros [...] We should really use ebay/aliexpress for these comparisons

Maybe you get a Roland cable clone on aliexpress for much less than 60 € ;-)
